core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -0,0 +1,135 @@
+        return Messages.DisablePluginCommand_ShortDescription();
+    }
+
+    @Override
+    protected void printUsageSummary(PrintStream stderr) {
+        stderr.println(Messages.DisablePluginCommand_PrintUsageSummary());
+    }
+
+    @Override
+    protected int run() throws Exception {
+        Jenkins jenkins = Jenkins.get();
+        jenkins.checkPermission(Jenkins.ADMINISTER);
+
+        PluginManager manager = jenkins.getPluginManager();
+        int result = 0;
+        boolean somePluginDisabled = false; // to ensure we restart when it needs to
+        for (String pluginName : pluginNames) {
+            switch (disablePlugin(manager, pluginName)) {
+                case DISABLED:
+                    if(!somePluginDisabled) somePluginDisabled = true;
+                    break;
+                case NOT_DISABLED_DEPENDANTS:
+                    if(result != 16) result = 16; // Custom error code to indicate one plugin, at least, can't be disabled
+            }
+        }
+
+        if (restart && somePluginDisabled) {
+            jenkins.safeRestart();
+        }
+
+        return result;
+    }
+
+    /**
+     * Try to disable a  plugin.
+     * @param manager The PluginManager.
+     * @param shortName The name of the plugin to disable.
+     * @return The result of the disabling of this plugin. See {@link DISABLING_STATUS}
+     * @throws IOException An exception disabling the plugin. See {@link PluginWrapper#disable()}
+     */
+    private DISABLING_STATUS disablePlugin(PluginManager manager, String shortName) throws IOException {
+        PluginWrapper plugin = manager.getPlugin(shortName);
+        if (plugin == null) {
+            throw new IllegalArgumentException(Messages.DisablePluginCommand_NoSuchPlugin(shortName)); // exit with 3
+        }
+
+        if (!plugin.isEnabled()) {
+            stdout.format(Messages.DisablePluginCommand_Already_Disabled(shortName));
+            stdout.println();
+            return DISABLING_STATUS.ALREADY_DISABLED;
+        }
+
+        Set<String> dependants = plugin.getDependants();
+        for (String dependant : dependants) {
+            PluginWrapper dependantPlugin = manager.getPlugin(dependant);
+            if (dependantPlugin != null && dependantPlugin.isEnabled()) {
+                // TO-DO: stdout or stderr? the process continues but as the result code is not 0, it looks like stderr
+                stderr.format(Messages.DisablePluginCommand_Plugin_Has_Dependant(shortName, dependant));
+                stderr.println();
+                return DISABLING_STATUS.NOT_DISABLED_DEPENDANTS;
+            }
+        }
+
+        plugin.disable();
+        stdout.format(Messages.DisablePluginCommand_Plugin_Disabled(plugin.getShortName()));
+        stdout.println();
+
+        return DISABLING_STATUS.DISABLED;
+    }
+
+    /**
+     * An enum to hold the status of a disabling action. To do it more reader-friendly.
+     */
+    private enum DISABLING_STATUS {
```

Comment:
Please follow Java convention, even enum must be CamelCased => `DisablingStatus`.

Reply:
Right! it slipped my mind